.travis.yml

```diff
@@ -8,12 +8,12 @@ env:
     - SHELL=zsh
 matrix:
   fast_finish: true
-  allow_failures:
-    - env: SHELL=zsh
 script:
   - |
     if ${LINT:-false}; then
       shellcheck -s sh bin/shpec  # 'sh' refers to POSIX
+    elif [ "$SHELL" = "zsh" ] ; then
```

Comment:
I would prefer not to have a separate 'test' command in the case of zsh, only separate setup. Something like:

```bash
if [ "$SHELL" = "zsh" ]; then
  disable -r end
fi

if ${LINT:-false}; then
# ... etc
else
  $SHELL bin/shpec
```
fi

N.B. We don't want to pass arguments to the `shpec` command, that would force us to update the arguments as we add test files (as is the case with this PR).